kani-compiler/src/intrinsics.rs

```diff
@@ -0,0 +1,359 @@
+    RotateLeft,
+    RotateRight,
+    RoundF32,
+    RoundF64,
+    SaturatingAdd,
+    SaturatingSub,
+    SinF32,
+    SinF64,
+    SimdAdd,
+    SimdAnd,
+    SimdDiv,
+    SimdRem,
+    SimdEq,
+    SimdExtract,
+    SimdGe,
+    SimdGt,
+    SimdInsert,
+    SimdLe,
+    SimdLt,
+    SimdMul,
+    SimdNe,
+    SimdOr,
+    SimdShl,
+    SimdShr,
+    SimdShuffle(&'a str),
+    SimdSub,
+    SimdXor,
+    SizeOfVal,
+    SqrtF32,
+    SqrtF64,
+    SubWithOverflow,
+    Transmute,
+    TruncF32,
+    TruncF64,
+    TypeId,
+    TypeName,
+    TypedSwap,
+    UnalignedVolatileLoad,
+    UncheckedDiv,
+    UncheckedRem,
+    Unlikely,
+    VolatileCopyMemory,
+    VolatileCopyNonOverlappingMemory,
+    VolatileLoad,
+    VolatileStore,
+    VtableSize,
+    VtableAlign,
+    WrappingAdd,
+    WrappingMul,
+    WrappingSub,
+    WriteBytes,
+    Unimplemented { name: &'a str, issue_link: &'a str },
+}
+
+impl<'a> Intrinsic<'a> {
+    pub fn from_str(intrinsic_str: &'a str) -> Self {
+        match intrinsic_str {
+            "add_with_overflow" => Self::AddWithOverflow,
+            "arith_offset" => Self::ArithOffset,
+            "assert_inhabited" => Self::AssertInhabited,
+            "assert_mem_uninitialized_valid" => Self::AssertMemUninitializedValid,
+            "assert_zero_valid" => Self::AssertZeroValid,
+            "assume" => Self::Assume,
+            name if name.starts_with("atomic_and") => {
+                Self::AtomicAnd(name.strip_prefix("atomic_and_").unwrap())
+            }
+            name if name.starts_with("atomic_cxchgweak") => {
+                Self::AtomicCxchgWeak(name.strip_prefix("atomic_cxchgweak_").unwrap())
+            }
+            name if name.starts_with("atomic_cxchg") => {
+                Self::AtomicCxchg(name.strip_prefix("atomic_cxchg_").unwrap())
+            }
+            name if name.starts_with("atomic_fence") => {
+                Self::AtomicFence(name.strip_prefix("atomic_fence_").unwrap())
+            }
+            name if name.starts_with("atomic_load") => {
+                Self::AtomicLoad(name.strip_prefix("atomic_load_").unwrap())
+            }
+            name if name.starts_with("atomic_max") => {
+                Self::AtomicMax(name.strip_prefix("atomic_max_").unwrap())
+            }
+            name if name.starts_with("atomic_min") => {
+                Self::AtomicMin(name.strip_prefix("atomic_min_").unwrap())
+            }
+            name if name.starts_with("atomic_nand") => {
+                Self::AtomicNand(name.strip_prefix("atomic_nand_").unwrap())
+            }
+            name if name.starts_with("atomic_or") => {
+                Self::AtomicOr(name.strip_prefix("atomic_or_").unwrap())
+            }
+            name if name.starts_with("atomic_singlethreadfence") => Self::AtomicSingleThreadFence(
+                name.strip_prefix("atomic_singlethreadfence_").unwrap(),
+            ),
+            name if name.starts_with("atomic_store") => {
```

Comment:
I like the macro approach! I will add it to the code.

Reply:
Actually, given that the code has changed quite a bit, I wonder if adding another macro is a good idea. I would be leaning to leaving this in if it is ok

Reply:
I think what I care most about is to avoid the unnecessary `unwrap`. I personally prefer macros to do it, but you can just as well use the if-let chain I suggested first.

Reply:
In case you are still wondering about this: I recommend using the `if let Some(...) = ...` chains.

Reply:
Added that ^